### PR TITLE
Update scopes documentation for Java and Android

### PR DESCRIPTION
**Pull request by designedbyz**
I was looking for information on the Sentry Kotlin extensions recently and as they're only significantly used for scopes found my way there. I found the documentation was a bit confusing, and decided to improve it. I also hid local scopes for Android since the documentation said that they are not supported there, and displayed the Kotlin Extensions information, as many Android apps use Kotlin and Coroutines (like the one I work on that uses Sentry!!). Finally, I moved the scope synchronization information out of the middle of the demonstration of setting and unsetting a user with `with-config`

I realize that the [contributing docs](https://docs.sentry.io/contributing/approach/sdk-docs/write-enriching-events/) say not to touch the common content, but that was very much needed, both to improve clarity and clean up the Android documentation.

### CODE REVIEW

**Comment by designedbyz** [comment]
Hello! I just wanted to bump this and see if there's anything you need from me to get this merged. Happy to make changes as needed!

**Comment by imatwawana** [comment]
> Hello! I just wanted to bump this and see if there's anything you need from me to get this merged. Happy to make changes as needed!

Hi @designedbyz, not yet. Once technical review is completed on this, we'll review it for writing/style and then there might be some feedback. Sorry about the delay.

File: src/includes/getting-started-install/android.mdx
Comment (marandaneto) [inline src/includes/getting-started-install/android.mdx:31]: This should not be here, since it only makes sense to use `SentryContext` if you have the concept of Local scopes, which is not the case for Android.

File: src/platforms/common/enriching-events/scopes.mdx
Comment (marandaneto) [inline src/platforms/common/enriching-events/scopes.mdx:94]: This is wrong, why is it only supported for Java?
See https://sentry-docs-git-fork-designedbyz-update-scopes.sentry.dev/platforms/apple/guides/ios/enriching-events/scopes/#local-scopes

File: src/platforms/common/enriching-events/scopes.mdx
Comment (marandaneto) [inline src/platforms/common/enriching-events/scopes.mdx:143]: This is probably wrong too, Kotlin coroutines is available even for iOS now: https://sentry-docs-git-fork-designedbyz-update-scopes.sentry.dev/platforms/apple/guides/ios/enriching-events/scopes/#kotlin-coroutines

**Comment by imatwawana** [comment]
@designedbyz, since there are a few concerns raised from technical review, I'll wait till those are resolved before I review for writing/style.

File: src/includes/getting-started-install/android.mdx
Comment (designedbyz) [reply src/includes/getting-started-install/android.mdx:31]: That totally makes sense, and I agree. The existing documentation both tells you how to use them and that you can't, so I erred on the side of improving what was there. Specifically, I removed directions on how to use them while leaving the mention that they can't be used as I wasn't completely comfortable with removing it entirely without a review. I can remove the mention entirely if that is preferred. 


| existing | this PR |
|---------|--------|
| image | image |

File: src/platforms/common/enriching-events/scopes.mdx
Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx:143]: Can you clarify? I'm confused about whether you mean the initial platform section was wrong, or if removing it was wrong. I'd think that if coroutines is available on iOS now that it might need mentioned?

File: src/includes/enriching-events/scopes/local-scope/android.mdx
Comment (designedbyz) [reply src/includes/enriching-events/scopes/local-scope/android.mdx:1]: Ok. My goal here was simply to clarify state as I understood it from the existing docs. Currently the Android docs tell you that Local Scopes can't be used and then proceed to tell you how to use them anyways, I thought it would be more clear to just leave it at saying that they can't be used. See my comment [here](https://github.com/getsentry/sentry-docs/pull/4747/files#r816456573) for a comparison of screenshots.

File: src/platforms/common/enriching-events/scopes.mdx
Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx]: Ooop, I think that's just a copy paste error on my part. 

Fixed in 1d1fbe03

**Comment by designedbyz** [comment]
> Hi @designedbyz, not yet. Once technical review is completed on this, we'll review it for writing/style and then there might be some feedback. Sorry about the delay.

Not a problem! I just wanted to check in. Thanks for getting eyes on this!

File: src/includes/getting-started-install/android.mdx
Comment (marandaneto) [reply src/includes/getting-started-install/android.mdx:31]: Yep, that would be ideal, I'd rather keep only for Java as it is right now https://docs.sentry.io/platforms/java/enriching-events/scopes/#kotlin-coroutines

File: src/platforms/common/enriching-events/scopes.mdx
Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx:143]: Sure, I just thought that the whole section of Kotlin Coroutines for iOS was visible because of the removal of L143, maybe it's not the case, but The Kotlin Coroutines should not be available for iOS, We have specific docs for Kotlin only https://docs.sentry.io/platforms/kotlin/
I'd not mix both contents yet while we don't offer official support for Kotlin multiplatform.

File: src/includes/enriching-events/scopes/local-scope/android.mdx
Comment (marandaneto) [reply src/includes/enriching-events/scopes/local-scope/android.mdx:1]: I'd go ahead with this change since it just improves the current state of the SDK, once we land https://github.com/getsentry/sentry-java/issues/1829
We can work on docs again.

File: src/includes/enriching-events/scopes/scope-synchronization/android.mdx
Comment (marandaneto) [reply src/includes/enriching-events/scopes/scope-synchronization/android.mdx:3]: Since this change isn't done yet, I'd leave it as it is and work with the migration docs once https://github.com/getsentry/sentry-java/issues/1914 lands.

File: src/platforms/common/enriching-events/scopes.mdx
Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx:143]: @marandaneto, got it. Fixed in 70231d38

File: src/includes/getting-started-install/android.mdx
Comment (designedbyz) [reply src/includes/getting-started-install/android.mdx:31]: @marandaneto, finally got back to this, done in 014eaa22

**Comment by designedbyz** [comment]
Hey @marandaneto, @bruno-garcia, any chance of an updated review?

**Comment by marandaneto** [comment]
> Hey @marandaneto, @bruno-garcia, any chance of an updated review?

src/platforms/common/enriching-events/attachments/index.mdx
  error: bad link platforms/android/enriching-events/scopes#local-scopes at line 72

Please fix the linking error, so I can check the preview and its changes, thanks :)

**Comment by designedbyz** [comment]
> Please fix the linking error, so I can check the preview and its changes, thanks :)

@marandaneto, finally got back to this, looks like attachments referenced local scopes, so I modified it for Android so that they didn't. 

Is there a way to run the build step that Vercel does locally? When I had that error, the docs would still build and display fine, and I certainly didn't expect changing info for scopes to break attachments!

**Comment by marandaneto** [comment]
> > Please fix the linking error, so I can check the preview and its changes, thanks :)
> 
> @marandaneto, finally got back to this, looks like attachments referenced local scopes, so I modified it for Android so that they didn't.
> 
> Is there a way to run the build step that Vercel does locally? When I had that error, the docs would still build and display fine, and I certainly didn't expect changing info for scopes to break attachments!

Vercel deploys automatically once we approve the CI to run, we can't do it locally, but you can build and run the docs locally, Just follow the dev. environment for the docs https://docs.sentry.io/contributing/environment/

**Comment by marandaneto** [comment]
> That said, what's up with that license compliance thing? Do I just need to rebase on the current main?

Not sure, this `Enforce License Compliance` is new for me too, @imatwawana can you help us out? thanks.

**Comment by designedbyz** [comment]
> Don't worry about it for now. It's an ongoing project of the open source team to automate licenses.

Got it, thanks!

**Comment by marandaneto** [comment]
> Yay, looks like the license check passed and Vercel deployed the most recent changes!

Thanks, I will review on Monday :)

**Comment by designedbyz** [comment]
> Thanks, I will review on Monday :)

Awesome, thank you!

File: src/includes/getting-started-install/android.mdx
Comment (marandaneto) [reply src/includes/getting-started-install/android.mdx:31]: Not sure if its not up to date (Vercel didn't deploy changes?), but I can still see it for Android https://sentry-docs-git-fork-designedbyz-update-scopes.sentry.dev/platforms/android/enriching-events/scopes/#kotlin-coroutines

**Comment by marandaneto** [comment]
@imatwawana it's ready for wording review, the only issue left is https://github.com/getsentry/sentry-docs/pull/4747#discussion_r841717679 from the technical point of view.

File: src/includes/getting-started-install/android.mdx
Comment (imatwawana) [inline src/includes/getting-started-install/android.mdx:31]:
```suggestion
If you are using Kotlin and coroutines in your Android project, you should read [Using Sentry SDK with Coroutines](/platforms/android/enriching-events/scopes/#kotlin-coroutines).
```

File: src/platforms/common/enriching-events/scopes.mdx
Comment (imatwawana) [inline src/platforms/common/enriching-events/scopes.mdx:43]: We use contractions wherever we can, but otherwise, this is a nice update.

```suggestion
of the box. It's unlikely that you'll interact with the hub directly unless you're
```

File: src/platforms/common/enriching-events/scopes.mdx
Comment (imatwawana) [inline src/platforms/common/enriching-events/scopes.mdx:44]:
```suggestion 
writing an integration or you want to create or destroy scopes. Scopes, on the
```

File: src/platforms/common/enriching-events/scopes.mdx
Comment (imatwawana) [inline src/platforms/common/enriching-events/scopes.mdx:108]: We should avoid starting the sentence with a lower case letter
```suggestion
On the other hand, `with-scope` creates a clone of the current scope, and the changes
```

File: src/platforms/common/enriching-events/scopes.mdx
Comment (imatwawana) [inline src/platforms/common/enriching-events/scopes.mdx:115]: Required spacing for note/alert components or the markdown within the text doesn't render properly. Use bold rather than italics to denote importance.

```suggestion

Any exceptions that occur within the callback function for `with-scope` will not be
caught, and all errors that occur will be silently ignored and **not** reported.

```

File: src/platforms/common/enriching-events/scopes.mdx
Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx]: Thanks! Haha, I think this was one of the reasons I started editing the docs.

Got it. Yeah, your style is a bit less formal than mine.

File: src/platforms/common/enriching-events/scopes.mdx
Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx:122]: Oh! with [this](https://github.com/getsentry/sentry-docs/pull/4747#discussion_r841714836) did you mean that you wanted to remove the mention of Kotlin as well as the local scopes stuff?

File: src/platforms/common/enriching-events/scopes.mdx
Comment (imatwawana) [reply src/platforms/common/enriching-events/scopes.mdx]: > Thanks! Haha, I think this was one of the reasons I started editing the docs.
> 
> Got it. Yeah, your style is a bit less formal than mine.

Yeah, we love a contraction around here 😄 but it's just as common to not use them. Thanks for all your work on this!

File: src/platforms/common/enriching-events/scopes.mdx
Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx:122]: Yes, for Android, since there's a global hub/scope, it does not make sense to use that integration, so we can remove from the android page.

Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx:122]: Got it, that makes sense. As a counter point though, I was confused that it was mentioned for Java and not Android, which is why I added it to begin with. What do you think about mentioning somewhere in the Android setup docs that the Kotlin extensions aren't needed for Android?

Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx:122]: Not sure, the content is not for Android at all, it even says:

> Sentry's SDK for Java stores the scope and the context in a thread-local variable

Instead of Sentry's SDK for Android, which is the case for all the other Android integrations, I guess explaining something from another platform would also make confusion.

Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx:122]: @designedbyz would you like to address this point so we can merge this? thanks.

Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx:122]: Sorry, too many irons in the fire. I should be able to address either later today or early tomorrow.

Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx]: @marandaneto done in 10b558ec8

**Review by marandaneto — APPROVED** [review]

File: src/platforms/common/enriching-events/scopes.mdx
Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx]: Agree, The goal is that `sentry-kotlin-extensions` get expanded and add extension functions to be "Kotlin friendly", but right now it only has the `SentryContext` and this only makes sense for non-Android.

E.g. https://github.com/getsentry/sentry-java/pull/1886/files
This would make sense for Android too.

Comment (marandaneto) [reply src/platforms/common/enriching-events/scopes.mdx]: I'd leave this specific part for another PR, this will get addressed automatically when we land the 1st Kotlin function in the `sentry-kotlin-extensions` library for Android, which is gonna be very soon :)
Thanks for the feedback.

Comment (designedbyz) [reply src/platforms/common/enriching-events/scopes.mdx]: > Agree, The goal is that sentry-kotlin-extensions get expanded and add extension functions to be "Kotlin friendly", but right now it only has the SentryContext and this only makes sense for non-Android.

Got it, that makes sense. 

> E.g. https://github.com/getsentry/sentry-java/pull/1886/files
This would make sense for Android too.

Oooh, that would be handy. 

> I'd leave this specific part for another PR, this will get addressed automatically when we land the 1st Kotlin function in the sentry-kotlin-extensions library for Android, which is gonna be very soon :)
Thanks for the feedback.

Awesome! That sounds good to me!

**Comment by marandaneto** [comment]
Thanks @designedbyz and sorry for the back and forth.

**Comment by designedbyz** [comment]
@marandaneto You're welcome, don't worry about it, I'm responsible for a bunch of it 😅  Thank you for reviewing!

**Pull request merged**
